application/app/Http/Controllers/DepartmentController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\StoreDepartmentRequest;
+use App\Http\Requests\UpdateDepartmentRequest;
+use App\Http\Resources\DepartmentResource;
+use App\Models\Department;
+use App\Models\Institution;
+use App\Policies\DepartmentPolicy;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Contracts\Database\Eloquent\Builder;
+use Illuminate\Http\Request;
+use Illuminate\Http\Resources\Json\ResourceCollection;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+use Throwable;
+
+class DepartmentController extends Controller
+{
+    public function index(): ResourceCollection
+    {
+        return DepartmentResource::collection($this->getBaseQuery()->get());
+    }
+
+    /**
+     * @throws AuthorizationException
+     * @throws Throwable
+     */
+    public function store(StoreDepartmentRequest $request): DepartmentResource
+    {
+        return DB::transaction(function () use ($request): DepartmentResource {
+            $this->authorize('create', Department::class);
```

Comment:
[non-blocking] Just a minor suggestion to move authorization check out of DB transaction. No need to start a DB transaction in case a user is not authorized.

Reply:
Well-spotted! I didn’t think of that.